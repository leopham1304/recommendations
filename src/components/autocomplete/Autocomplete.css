.autocomplete {
  font-family: "Roboto", sans-serif;
  font-size: 18px;
  border: none;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
  color: #484848;
  width: 70vw;
  max-width: 800px;
}

.autocomplete input {
  font-family: "Roboto", sans-serif;
  font-size: 20px;
  font-weight: 600;
  background: url(../../assets/images/search.svg) #f8f8f8 no-repeat scroll 15px
    center;
  width: 100%;
  padding: 15px 55px;
  color: #484848;
  box-sizing: border-box;
  outline: none;
  border: 1px solid #e3dfdf;
}

.autocomplete input:placeholder-shown {
  border-radius: 4px;
}

.autocomplete input:not(:placeholder-shown) {
  border-radius: 4px 4px 0px 0px;
}

.autocomplete ul {
  list-style-type: none;
  text-align: left;
  margin: 0;
  padding: 0;
  border: 1px solid #ebebeb;
  border-radius: 0px 0px 4px 4px;
}

.autocomplete li {
  padding: 18px 20px;
  cursor: pointer;
  box-sizing: content-box;
  background-color: #fff;
}

.autocomplete li:hover {
  background-color: #f8f8f8;
}
