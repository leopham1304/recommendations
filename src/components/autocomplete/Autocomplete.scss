.autocomplete {
  font-size: 18px;
  border: none;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
  width: 70vw;
  max-width: 600px;

  input {
    background: url(../../assets/images/search.svg) #f8f8f8 no-repeat scroll
      15px center;
    border: 1px solid #e3dfdf;
    color: #484848;
    font-size: 20px;
    font-weight: 600;
    width: 100%;
    padding: 15px 55px;
    box-sizing: border-box;
    outline: none;
  }

  input:placeholder-shown {
    border-radius: 4px;
  }

  input:not(:placeholder-shown) {
    border-radius: 4px 4px 0px 0px;
  }

  ul {
    border: 1px solid #ebebeb;
    list-style-type: none;
    text-align: left;
    margin: 0;
    padding: 0;
    border-radius: 0px 0px 4px 4px;
  }

  li {
    background-color: #fff;
    padding: 18px 20px;
    cursor: pointer;
    box-sizing: content-box;
  }

  li:hover {
    background-color: #f8f8f8;
  }
}

.dark-mode {
  input {
    background: url(../../assets/images/search.svg) #0e0e0e no-repeat scroll
      15px center;
    border: 1px solid #565656;
    color: #cdcdcd;
  }
  ul {
    border: 1px solid #565656;
  }
  li {
    background-color: #1b1b1b;
    color: #cdcdcd;
  }
  li:hover {
    background-color: #565656;
  }
}
