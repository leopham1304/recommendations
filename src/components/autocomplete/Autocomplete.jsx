import React, { memo } from "react";
import { useAutoComplete } from "./Autocomplete.utils";
import SuggestionItem from "./suggestion-item/SuggestionItem";
import "./Autocomplete.scss";

const Autocomplete = () => {
  const {
    handleSearchChange,
    handleSelect,
    search,
    suggestions,
  } = useAutoComplete();

  return (
    <div className="autocomplete">
      <input
        onChange={handleSearchChange}
        placeholder="Try Vietnam"
        value={search}
      />
      {!!suggestions.length && (
        <ul>
          {suggestions.map((suggestion) => (
            <SuggestionItem
              key={suggestion}
              onSelect={handleSelect}
              suggestion={suggestion}
            />
          ))}
        </ul>
      )}
    </div>
  );
};

export default memo(Autocomplete);
