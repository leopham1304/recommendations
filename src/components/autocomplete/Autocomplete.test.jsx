import React from "react";
import Autocomplete from "./Autocomplete";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

// Cache original functionality
const realUseState = React.useState;

// Mock useState before rendering component
jest
  .spyOn(React, "useState")
  .mockImplementationOnce(() => realUseState("Ho"))
  .mockImplementationOnce(() => realUseState(["Ho Chi Minh"]));

describe("Autocomplete", () => {
  const wrapper = shallow(<Autocomplete darkMode={false} />);

  it("should render the Autocomplete Component correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
