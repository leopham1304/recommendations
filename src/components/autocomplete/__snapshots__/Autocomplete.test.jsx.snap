// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete should render the Autocomplete Component correctly 1`] = `
<div
  className="autocomplete"
>
  <input
    onChange={[Function]}
    placeholder="Try Vietnam"
    value="Ho"
  />
  <ul>
    <Memo(SuggestionItem)
      key="Ho Chi Minh"
      onSelect={[Function]}
      suggestion="Ho Chi Minh"
    />
  </ul>
</div>
`;
